Settings/TrackingConfigPanel: same tracking intervals for Livetrack24 as for SkyLines

sandrinr opened the pull request:
<!--

Thank you for your interest in contributing to XCSoar! Please read the
following information to make it easier for us to review your changes.

We appreciate if you make sure to:

  - Document the changes (in the NEWS.txt file, and the manual when relevant)
  - Enable maintainer edits[1] (in case we need to help with the PR)
  - Check your commits and their messages (so they're all tidy and coherent)

[1] https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork

-->


Brief summary of the changes
----------------------------

Use the existing tracking intervals list for SkyLines also for LiveTrack24.
Currently LiveTrack24 allows tracking intervals between 5 seconds and 1 hour using 5 second increments.

Intent: Allow LiveTrack24 tracking intervals smaller than 5 seconds.

Related issues and discussions
------------------------------

<!--
Please link any relevant issues or forum posts here, for reference.

If this PR resolves an existing issue, please write "Closes #1234" so that
the issue is closed automatically when this PR is merged.
-->

none


Caveats
--------

Users with an existing setting of `TrackingInterval` outside of the values specified in the `tracking_intervals` array (e.g. "1800") will have their tracking interval set to 1 second.

MaxKellermann commented:
> Users with an existing setting of TrackingInterval outside of the values specified in the tracking_intervals array (e.g. "1800") will have their tracking interval set to 1 second.

That's not acceptable. Since the LT24 protocol is heavily bloated, this will cause big amounts of mobile traffic.

sandrinr commented:
> That's not acceptable. Since the LT24 protocol is heavily bloated, this will cause big amounts of mobile traffic.

Understandable. What would be your minimal acceptance criteria?
Initializing it to a value close to the original setting value?
Would you like it more me creating a new enum for LT24 (instead of using the one for SkyLines) holding all 5 second increments up to 60 minutes in order to be compatible with every possible old setting value?

More generally, do you have another idea to allow sub 5 second intervals not involving an enum?

MaxKellermann commented:
It is not acceptable to break existing configurations, even less so if it means the user has to pay money for unwanted internet traffic. All changes to configuration must be migrated properly; I guess rounding slightly up or down to the nearest supported value would be acceptable, but not resetting the configuration.
Neither the integer DataField nor the EnumDataField are a good choice for this setting. Maybe we need a new DataField class for configuring a time interval, with a number of suggested presets, but if an existing value is not in those presets, it will appear anyway in the list.

sandrinr commented:
Updated the branch. Went the route of adding some more values to `tracking_intervals` and added a method to find the closest tracking interval setting given an input setting.
